Added Demographics Section

noranda opened the pull request:
Also cleaned up some code and added default values to the form field components.

cmoesel commented:
This is looking really good, @noranda!  It's fun reviewing this stuff and learning from you!  You've got a nice code style!  In testing this, I have a few questions/comments for your consideration:

1. Should we consider adding a hint on the date fields that indicates the expected format? For people typing the date, it isn't clear until they get validation errors.
2. I typed in a date and then deleted it (using backspace) and the app crashed:
![image](https://user-images.githubusercontent.com/2278253/77235041-0235e700-6b89-11ea-8e6a-4a89e9f8e367.png)
3. The Age field allows me to type in a non-number, but then renders weird:
![image](https://user-images.githubusercontent.com/2278253/77235110-80928900-6b89-11ea-8d6c-5061b18c858c.png)
vs.
![image](https://user-images.githubusercontent.com/2278253/77235116-90aa6880-6b89-11ea-86a2-0fbc57943f87.png)
(Ignore the weird blue reindeer icon -- that's my Dashlane autofill widget)
4. Minor and subjective comment: I feel like the Race and Ethnicity fields should be flipped.

cmoesel commented on src/components/forms/DateField/DateField.js:
```diff
@@ -1,21 +1,20 @@
   const styles = useStyles();
   const { setForm } = useForm();
-  const [selectedDate, setDate] = useState(moment());
+  const [selectedDate, setDate] = useState(defaultValue);
```
Now that we're supporting `defaultValue`, shouldn't we also be setting it on the `form` here?  If the user chooses to accept the default value, then the `handleDateChange` function will never get called and the date will never actual be put in the `form` field.

noranda replied:
Good observation! For this and a few other annoying reasons, I think we should use a library called [Formik](https://jaredpalmer.com/formik/) which takes care of keeping track of values (including initial values), errors, submission, and at some point validation if we like. I noticed our components were getting a little unwieldy and thought I'd see how much work it would be to convert now rather than later (turns out not too much work). @radamson and I discussed this possibility so my next PR should fix this issue.

cmoesel commented on src/components/forms/SelectBox/SelectBox.js:
```diff
@@ -3,28 +3,28 @@ import TextField from '@material-ui/core/TextField';
+function SelectBox({ name, label, defaultValue, options }) {
   const { setForm } = useForm();
-  const [selectedItem, setItem] = useState();
+  const [selectedItem, setItem] = useState(defaultValue);
```
Same comment re: `defaultValue` as above.

cmoesel replied:
I won't repeat myself on the others -- but just pretend like I said this everywhere we have a`defaultValue`. ;-)

noranda replied:
Same response. ;)

cmoesel commented on src/components/sections/DemographicsSection/DemographicsSection.js:
```diff
@@ -0,0 +1,79 @@
+  { value: 'hispanicLatino', label: 'Hispanic / Latino' },
+  { value: 'nonHispanicLatino', label: 'Non-Hispanic / Latino' },
+  { value: 'notSpecified', label: 'Not Specified' }
+];
```
I know this is following the pattern I did in the `SelectBox` -- but I was wondering -- should we simplify this to:
```javascript
const ethnicityOptions = {
  hispanicLatino: 'Hispanic / Latino',
  nonHispanicLatino: 'Non-Hispanic / Latino' ,
  notSpecified: 'Not Specified'
};
```
The upside is it is definitely more succinct.  The downside is that if we ever wanted to add more information for each item, it doesn't support that as easily.  Anyway, it's probably fine as-is, but I thought maybe I'd think out loud.

noranda replied:
I like it, but I won't change it in this PR because I'm pretty sure it'll blow up my next PR haha.

noranda commented:
@cmoesel Thanks for the comments!

1. Yes, definitely, and I've done this in my next branch.
2. Fixed this bug in my next branch.
3. Fixed this bug in my next branch (are you sensing a pattern?)
4. Maybe? They're in weird places on the paper form and the ordering actually makes things easier if the user selects "Other" for race, which will pop up another text field to specify the race.
